### ISSUE TITLE
OutdatedDocumentation triggered for @param for "private val" constructor parameter

### ISSUE THREAD

**Issue by mtrewartha** [detekt/detekt#5298]
## Expected Behavior
When I have a class with a public constructor that takes a parameter marked as `private val`, I should be able to document that parameter with `@param`.

## Observed Behavior
The `OutdatedDocumentation` rule fails.

## Steps to Reproduce
1. Create a class like so:
    ```
    /**
     * Some docs
     * 
     * @constructor Creates a new instance
     * 
     * @param someParam Some parameter for the constructor
     */
    class SomeClass(
        private val someParam: Int
    ) {
        fun doSomething(): Int = someParam + 1
    }
    ```
2. Make sure the `OutdatedDocumentation` rule is active.
2. Run Detekt.

## Context
We have a library module in our codebase that we're trying to thoroughly document, but we've run in to this a few times now. 

## Your Environment
* Version of detekt used: 1.21.0
* Version of Gradle used (if applicable): 7.5.1
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: macOS Ventura 13.0 Beta (22A5342f) on an Apple M1 Pro
* Link to your project (if it's a public repository): Sorry, it's private.

**Comment by schalkms** [comment]
You can set the rule config `"allowParamOnConstructorProperties" to "true"`. After that, your code is not flagged anymore.

**Comment by mtrewartha** [comment]
Oh, my bad for not realizing that in the first place. Thanks for the help!

**Comment by schalkms** [comment]
You are welcome!
